frigate/comms/zmq_proxy.py

```diff
@@ -0,0 +1,100 @@
+"""Facilitates communication over zmq proxy."""
```

Comment:
I refactored a lot of the pub/sub code here to make it easier to extend and add new things later. This file contains a single proxy implementation and base classes for `Publisher` and `Subscriber` which `events_updater.py` and `detections_updater.py` were refactored to use.